StackExchange/Stacks
# feat(button): add "featured" variant

[STACKS-680](https://stackoverflow.atlassian.net/browse/STACKS-680)

## TODO

> [!NOTE]
> I've opened this PR up for review but there's a couple of outstanding items that need to be resolved:

- [x] Verify name "featured"
- [ ] Write description in docs

---

This PR adds a "featured" (purple) variant to the button component.

<details>
<summary>Screenshots</summary>

## Light

![image](https://github.com/user-attachments/assets/3387df5f-b0cc-4ce4-9081-1c49a06fc780)

## Dark

![image](https://github.com/user-attachments/assets/299cef03-ebcc-4fa6-9ca3-92a34d4ea0a9)

## Light high contrast

![image](https://github.com/user-attachments/assets/d5847be4-d2a6-478d-b864-9a2e528d2717)

## Dark high contrast

![image](https://github.com/user-attachments/assets/2141d4f0-7ec2-4ac1-8f56-d25d87391f95)

</details>

## Testing

1. Go to `/product/components/buttons/#featured`
2. Verify the newly added _featured_ buttons look and act as expected

## Notable changes

### High contrast filled background color

~~To meet accessibility requirements, the featured filled variant background color has been set to `--purple-500` (normally `--purple-400`).~~

We've modified the `purple-400` light HC mode color instead. See https://github.com/StackExchange/Stacks/pull/1857#pullrequestreview-2466079719

### Borders on filled, selected buttons

This PR includes a minor fix to button border colors when hovered and filled and/or selected. Previously, the value of `--bu-bc` (`currentColor`) was applied and now the border color applied is equal to `--bu-bg`, ensuring the border and button background are unified.

### Button group visual regression test images

There are six visual regression tests (all on `s-btn-group-highcontrast-*` in Chromium runs) that have been updated. They are visually identical to me but there could be extremely slight differences due to https://github.com/StackExchange/Stacks/pull/1857/commits/838e25ad90ab4b68e52b54c2e89e2b8b43bf4054.
